### PR TITLE
Enable simple ROW support 

### PR DESCRIPTION
**Pull request by jarodmeng**
Enable simple ROW support by translating a ROW column in Presto to a list-column of named lists (very similar to MAP) in R.

Unlike MAP, the subfield names are not transmitted in the data JSON, so we need to grab the subfield names from the schema JSON and manually set them for any ROW columns.

I added  unit tests in test-data_types.R to ensure the translation works as intended.

All tests passed locally.

```
==> Rcpp::compileAttributes()

* Updated src/RcppExports.cpp
* Updated R/RcppExports.R

==> Sourcing R files in 'tests' directory

══ Skipped tests ═══════════════════════════════════════════════════════════════
• non utf8 locale (2)

[ FAIL 0 | WARN 0 | SKIP 2 | PASS 440 ]

Tests complete
```

This PR alleviates #118 .

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -29,7 +29,8 @@ Imports:
     stringi,
     stats,
     Rcpp (>= 0.12.7),
-    utils
+    utils,
+    purrr
```
Comment (leepface) [inline DESCRIPTION:33]: Can we do this without purrr? I'm hesitant to add any dependencies.

Comment (jarodmeng) [reply DESCRIPTION:32]: For this simple ROW support, I think I can use `lappy()` in place of `purrr::map()`. However, I already have a  rough plan on how to make ARRAY/MAP/ROW support more comprehensive. In that plan, I use extensively functions from `purrr` as it is built to better deal with lists, especially nested lists (which are how the JSON objects of both the schema and data are).

I want to understand a bit better why you prefer not to add the dependency.
1) quality of the `purrr` package
2) increased size of RPresto
3) speed impact on functions in RPresto
4) something else

IMHO, `purrr` has become a quite fundamental part of the R and tidyverse ecosystem (literally 100+ CRAN packages now import it as a dependency, see https://cran.r-project.org/web/packages/purrr/index.html).

Comment (leepface) [reply DESCRIPTION:32]: My main concern is it historically had an unstable API. If we use core features that are stable (and unlikely to be deprecated on a whim in the future), it's fine.

Comment (jarodmeng) [reply DESCRIPTION:32]: That's a valid concern when it comes to adding a new dependency. I've followed `purrr`'s development for 5+  years now. I think most functions in the package have been stable. The ones I used (i.e., `purrr::map_*()` and `purrr::modify_*()`) have been the core of `purrr` and should be kept stable even when a new version is released. I will refrain from using those with "questioning" lifecycle.

Comment (leepface) [reply DESCRIPTION:32]: Sounds good.

**Comment by jarodmeng** [comment]
Any other follow-ups on this PR?

**Pull request merged**